src/CoreFoundation/CFProxySupport.cs

```diff
@@ -580,16 +580,23 @@ public static CFProxySettings GetSystemProxySettings ()
 			return new CFProxySettings (dict);
 		}
 		
-#if notyet
-		// FIXME: These require bindings for CFRunLoopSource and CFStreamClientContext
-		
 		public delegate void CFProxyAutoConfigurationResultCallback (IntPtr client, NSArray proxyList, NSError error);
+		static CFProxyAutoConfigurationResultCallback static_AutoConfigurationResult = TrampolineAutoConfigurationResult;
+
+		[MonoPInvokeCallback (typeof (CFProxyAutoConfigurationResultCallback))]
+		static void TrampolineAutoConfigurationResult (IntPtr block, NSArray proxyList, NSError error)
+		{
+			var del = BlockLiteral.GetTarget<Action<NSArray, NSError>> (block);
+			if (del != null) {
+				del (proxyList, error);
+			}
+		}
 		
 		[DllImport (Constants.CFNetworkLibrary)]
-		extern static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationScript (
+		extern unsafe static /* CFRunLoopSourceRef __nonnull */ IntPtr CFNetworkExecuteProxyAutoConfigurationScript (
 			/* CFStringRef __nonnull */ IntPtr proxyAutoConfigurationScript,
 			/* CFURLRef __nonnull */ IntPtr targetURL,
-			/* CFProxyAutoConfigurationResultCallback __nonnull */ IntPtr cb,
+			/* CFProxyAutoConfigurationResultCallback __nonnull */ void* cb,
 			/* CFStreamClientContext * __nonnull */ IntPtr clientContext);
 		
 		public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (NSString proxyAutoConfigurationScript, NSUrl targetURL, CFProxyAutoConfigurationResultCallback resultCallback, CFStreamClientContext clientContext)
```

Comment:
A signature like this would be much nicer:

```csharp
delegate void AutoConfigurationResultCallback (NSArray proxyList, NSError error);
public static CFRunLoopSource ExecuteProxyAutoConfigurationScript (string proxyAutoConfigurationScript, NSUrl targetUrl, AutoConfigurationResultCallback callback)
{
    // ...
}
```